Create net.poedit.Poedit.json

bil-elmoussaoui opened the pull request:
Not sure about two things:
- Does Poedit really require the gnome sdk?
- Should I add gtkspell as a module or it's already in the gnome sdk?

nedrichards commented:
I'd say, looking briefly at it that the FDo SDK is probably most relevant for this app.

You should add gtkspell as a module: https://github.com/search?q=org%3Aflathub+gtkspell&type=Code has some examples of other apps doing that.

nedrichards commented:
bot, build net.poedit.Poedit

nedrichards commented:
bot, build net.poedit.Poedit

bil-elmoussaoui commented:
Alright,  I have added gtkspell and switched to freedesktop sdk, I didn't understand the issue with wxWidgets not compiling for no reason :/

nedrichards commented:
You need to autoreconf wxWidgets to work on aarch64 - since the tarball was released before 64 bit arm existed. Here's an example: https://github.com/flathub/org.DolphinEmu.dolphin-emu/blob/98d37a870252a19e8e31e0b3841eb24788904526/org.DolphinEmu.dolphin-emu.json#L20

Although the readme says greater than or equal to 3.0.3? It may be better to use 3.1 as in https://github.com/flathub/com.bixense.PasswordCalculator/blob/master/com.bixense.PasswordCalculator.json#L14

You also have a compilation issue with lucene++

bil-elmoussaoui commented:
Alright, i'm going to fix the wxwidgets issue. 
For the lucene++ one; I guess it's due to this 
```
[ 54%] Linking CXX executable deletefiles
/usr/lib/gcc/x86_64-unknown-linux/6.2.0/../../../../x86_64-unknown-linux/bin/ld: cannot open output file deletefiles: Is a directory
collect2: error: ld returned 1 exit status
make[2]: *** [src/demo/CMakeFiles/deletefiles.dir/build.make:107: src/demo/deletefiles] Error 1
make[1]: *** [CMakeFiles/Makefile2:504: src/demo/CMakeFiles/deletefiles.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 54%] Linking CXX executable indexfiles
/usr/lib/gcc/x86_64-unknown-linux/6.2.0/../../../../x86_64-unknown-linux/bin/ld: cannot open output file indexfiles: Is a directory
collect2: error: ld returned 1 exit status
make[2]: *** [src/demo/CMakeFiles/indexfiles.dir/build.make:107: src/demo/indexfiles] Error 1
make[1]: *** [CMakeFiles/Makefile2:541: src/demo/CMakeFiles/indexfiles.dir/all] Error 2
Scanning dependencies of target lucene++-contrib
[ 55%] Linking CXX executable searchfiles
/usr/lib/gcc/x86_64-unknown-linux/6.2.0/../../../../x86_64-unknown-linux/bin/ld: cannot open output file searchfiles: Is a directory
collect2: error: ld returned 1 exit status
make[2]: *** [src/demo/CMakeFiles/searchfiles.dir/build.make:107: src/demo/searchfiles] Error 1
make[1]: *** [CMakeFiles/Makefile2:467: src/demo/CMakeFiles/searchfiles.dir/all] Error 2
```
It seems to be due to the demo files. I will see if I can disable them 👍

nedrichards commented:
bot, build net.poedit.Poedit

bil-elmoussaoui commented:
I have reverted back to gnome sdk, otherwise gtkspell adds a bunch of other requirements.......
I have also fixed the lucene++ issues by disabling tests and demos by default, disabled "uneeded" wxwidgets that shouldn't be used by poedit anyway. 

But I'm still getting an issue with building Poedit itself
```
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether make supports nested variables... (cached) yes
checking for install location... /app
checking for gawk... (cached) gawk
checking whether ln -s works... yes
checking for ranlib... ranlib
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking how to run the C++ preprocessor... g++ -E
checking whether g++ supports C++14 features by default... yes
checking for boostlib >= 1.60... yes
checking whether the Boost::System library is available... yes
configure: error: Could not find a version of the library!
Error: module poedit: Child process exited with code 1
```

nedrichards commented:
The wxwidgets still needs the reconfigure mentioned above to work on aarch64 as well.

bil-elmoussaoui commented:
@nedrichards Fixed that! Any help on how to deal with the poedit build issue would be very welcome

Barthalion commented:
It says this at start of the build:
```
    * debug build:                    no
    * language detection:             no
    * crowdin integration:            no



    !!! WARNING !!!

    Your are building a limited version of Poedit without some important
    features (see above). This makes Poedit harder to use and is strongly
    advised against.

    !!! WARNING !!!`
```

No idea if it's something worth looking at.

bot, build net.poedit.Poedit

bil-elmoussaoui commented:
@Barthalion Haven't seen the config opt for boost libdir! Thanks!
I will try building this locally and see if everything works fine

bil-elmoussaoui commented:
Don't even know why poedit needs Network access? :p 
` ``
Debug: Failed to connect to session manager: Could not open network socket
```
It also requires dconf permissions :)
```
(poedit:3): dconf-WARNING **: failed to commit changes to dconf: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown
```
I have added both in my latest commit.

bil-elmoussaoui commented:
It seems that the two dependencies needed by Poedit (reported by @Barthalion 's comment) are 
- https://github.com/CLD2Owners/cld2 which doesn't have any release tarball...
- https://github.com/Microsoft/cpprestsdk which is used to with online features
So I'm going to add both. 
But still, Poedit crashes without any error message. I will enable poedit debug and see what happens

vslavik commented on net.poedit.Poedit.json:
```diff
@@ -0,0 +1,99 @@
+            ],
+            "sources": [{
+                    "type": "archive",
+                    "url": "https://github.com/wxWidgets/wxWidgets/releases/download/v3.1.0/wxWidgets-3.1.0.tar.bz2",
```
Using wx-3.1.0, a year-old *unstable development snapshot*, is a very bad idea — it is known buggy. Reasonable choices are latest stable or the submodule from Poedit's own tree.

bil-elmoussaoui replied:
Should I switch to 3.0.3?

bil-elmoussaoui replied:
Plus would like to hear your opinion about including cld2 & c++ rest sdk

vslavik replied:
The warning you quoted is pretty clear on my opinion, I would think: packaging degraded versions is frowned upon.

bil-elmoussaoui commented:
I have replaced wxWidgets with 3.0.3 instead of 3.1.0. Which fixes a crash whenever I tried to use the application.

The application works correctly as far as I can tell. Next step would be either we should include C++ REST SDK and cdl2

nedrichards commented:
CLD2 is basically included as is in Chrome as I understand, so I'm less worried about taking a snapshot from there than I otherwise would be - but overall we should include both - this is what the app says shows it off to it's best - then we should have it.

bil-elmoussaoui commented:
Alright, I will do some testing and push the changes by this evening

vslavik commented:
Sorry for jumping in late, I didn't expect this much activity here. As the Poedit maintainer, I have my own plans for shipping official Snap/Flatpak packages soon-ish. I'd much prefer if this was *not* merged, or if you're going to include it, I'd at least appreciate if the package was clearly labeled as unofficial and didn't use "my" `net.poedit.*` namespace for the app-id — so that it doesn't conflict with the official flatpak and cause a giant mess.

bil-elmoussaoui commented:
@vslavik the flatpak package was just an attempt to learn more how things works. You can be added as a collaborator once this PR is merged and you can do whatever modifications you wish. Otherwise I can close this PR and you can push yours whenever you want ;)

vslavik commented:
> a collaborator once this PR is merged and you can do whatever modifications you wish

If you're OK with that, then sure, I'm OK with it.

bil-elmoussaoui commented:
Yeah definitely. There's still a small list of issues to fix. Is there a way to talk about this and see if there's something that should be patched upstream?

nedrichards commented:
@vslavik a key tenet of flathub is that 'upstream owns their app' so absolutely you'd have full access. We think that using flathub makes it much easier for developers to get involved with flatpak (we handle repo management, building into arm and x86 etc.) and you'll get the (fairly rudimentary currently) download stats.

nedrichards commented:
bot, build net.poedit.Poedit

nedrichards commented:
@bil-elmoussaoui I assume that opening a ticket upstream is best for talking about upstream fixes. Is there anything more that you plan to do here?

nedrichards commented:
No - once it's in it's exposed to everyone. You can work on things then in PRs which aren't shipped but the core app is expected to work.

That's fine though - just push changes and one of the admins will trigger a rebuild for you.

nedrichards commented:
bot, build net.poedit.Poedit

bil-elmoussaoui commented:
Poedit now works without any issue! We should still add the two missing libraries! 
@vslavik What do you think?

nedrichards commented:
I've added the first missing library, and a currently non working version of cld2. I think I'm putting it in the right place (looks like you're meant to vendor it in to the main build. But it's not picking it up. Patches welcome.

nedrichards commented:
bot, build net.poedit.Poedit

bil-elmoussaoui commented:
bot, build net.poedit.Poedit

bil-elmoussaoui commented:
Fixed the cld2 build using the debian CMakeLists.txt file, it works here!
Thanks for adding cpprestsdk ! We can say that's ready now :)

nedrichards commented:
bot, build net.poedit.Poedit

nedrichards commented:
https://flathub.org/builds/#/builders/1/builds/2315

I think this is going to fail building on arm. If it does - can you put a flathub.json banning arm in and I'll merge after that and work on fixing it in a PR.

bil-elmoussaoui commented:
As you expected, it fails on arm. I will add a flathub.json right now :+1:

nedrichards commented:
Repository has been created: https://github.com/flathub/net.poedit.Poedit
